// simplebar-angular/esm5/lib/simplebar-angular.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, ElementRef, ViewChild } from '@angular/core';
import SimpleBar from 'simplebar';
var SimplebarAngularComponent = /** @class */ (function () {
    function SimplebarAngularComponent() {
    }
    /**
     * @return {?}
     */
    SimplebarAngularComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () { };
    /**
     * @return {?}
     */
    SimplebarAngularComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        this.SimpleBar = new SimpleBar(this.simplebar.nativeElement, this.options ? this.options : {});
    };
    SimplebarAngularComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-simplebar',
                    template: "<div data-simplebar #simplebar style=\"height:100%\">\r\n  <div class=\"simplebar-wrapper\">\r\n    <div class=\"simplebar-height-auto-observer-wrapper\">\r\n      <div class=\"simplebar-height-auto-observer\"></div>\r\n    </div>\r\n    <div class=\"simplebar-mask\">\r\n      <div class=\"simplebar-offset\">\r\n        <div class=\"simplebar-content-wrapper\">\r\n          <div class=\"simplebar-content\">\r\n            <ng-content></ng-content>\r\n          </div>\r\n        </div>\r\n      </div>\r\n    </div>\r\n    <div class=\"simplebar-placeholder\"></div>\r\n  </div>\r\n  <div class=\"simplebar-track simplebar-horizontal\">\r\n    <div class=\"simplebar-scrollbar\"></div>\r\n  </div>\r\n  <div class=\"simplebar-track simplebar-vertical\">\r\n    <div class=\"simplebar-scrollbar\"></div>\r\n  </div>\r\n</div>\r\n",
                    styles: ["[data-simplebar]{position:relative;-webkit-box-orient:vertical;-webkit-box-direction:normal;flex-direction:column;flex-wrap:wrap;-webkit-box-pack:start;justify-content:flex-start;align-content:flex-start;-webkit-box-align:start;align-items:flex-start}.simplebar-wrapper{overflow:hidden;width:inherit;height:inherit;max-width:inherit;max-height:inherit}.simplebar-mask{direction:inherit;position:absolute;overflow:hidden;padding:0;margin:0;left:0;top:0;bottom:0;right:0;width:auto!important;height:auto!important;z-index:0}.simplebar-offset{direction:inherit!important;box-sizing:inherit!important;resize:none!important;position:absolute;top:0;left:0;bottom:0;right:0;padding:0;margin:0;-webkit-overflow-scrolling:touch}.simplebar-content-wrapper{direction:inherit;box-sizing:border-box!important;position:relative;display:block;height:100%;width:auto;visibility:visible;overflow:auto;max-width:100%;max-height:100%}.simplebar-content:after,.simplebar-content:before{content:\" \";display:table}.simplebar-placeholder{max-height:100%;max-width:100%;width:100%;pointer-events:none}.simplebar-height-auto-observer-wrapper{box-sizing:inherit!important;height:100%;width:100%;max-width:1px;position:relative;float:left;max-height:1px;overflow:hidden;z-index:-1;padding:0;margin:0;pointer-events:none;-webkit-box-flex:inherit;flex-grow:inherit;flex-shrink:0;flex-basis:0}.simplebar-height-auto-observer{box-sizing:inherit;display:block;opacity:0;position:absolute;top:0;left:0;height:1000%;width:1000%;min-height:1px;min-width:1px;overflow:hidden;pointer-events:none;z-index:-1}.simplebar-track{z-index:1;position:absolute;right:0;bottom:0;pointer-events:none;overflow:hidden}[data-simplebar].simplebar-dragging .simplebar-content{pointer-events:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-user-select:none}[data-simplebar].simplebar-dragging .simplebar-track{pointer-events:all}.simplebar-scrollbar{position:absolute;right:2px;width:7px;min-height:10px}.simplebar-scrollbar:before{position:absolute;content:\"\";background:#000;border-radius:7px;left:0;right:0;opacity:0;-webkit-transition:opacity .2s linear;transition:opacity .2s linear}.simplebar-track .simplebar-scrollbar.simplebar-visible:before{opacity:.5;-webkit-transition:opacity linear;transition:opacity linear}.simplebar-track.simplebar-vertical{top:0;width:11px}.simplebar-track.simplebar-vertical .simplebar-scrollbar:before{top:2px;bottom:2px}.simplebar-track.simplebar-horizontal{left:0;height:11px}.simplebar-track.simplebar-horizontal .simplebar-scrollbar:before{height:100%;left:2px;right:2px}.simplebar-track.simplebar-horizontal .simplebar-scrollbar{right:auto;left:0;top:2px;height:7px;min-height:0;min-width:10px;width:auto}[data-simplebar-direction=rtl] .simplebar-track.simplebar-vertical{right:auto;left:0}.hs-dummy-scrollbar-size{direction:rtl;position:fixed;opacity:0;visibility:hidden;height:500px;width:500px;overflow-y:hidden;overflow-x:scroll}:host{display:block}"]
                }] }
    ];
    /** @nocollapse */
    SimplebarAngularComponent.ctorParameters = function () { return []; };
    SimplebarAngularComponent.propDecorators = {
        options: [{ type: Input }],
        simplebar: [{ type: ViewChild, args: ['simplebar', { static: true },] }]
    };
    return SimplebarAngularComponent;
}());
export { SimplebarAngularComponent };
if (false) {
    /** @type {?} */
    SimplebarAngularComponent.prototype.options;
    /** @type {?} */
    SimplebarAngularComponent.prototype.simplebar;
    /** @type {?} */
    SimplebarAngularComponent.prototype.SimpleBar;
}
/**
 * @record
 */
export function Options() { }
if (false) {
    /** @type {?|undefined} */
    Options.prototype.autoHide;
    /** @type {?|undefined} */
    Options.prototype.scrollbarMinSize;
    /** @type {?|undefined} */
    Options.prototype.classNames;
    /** @type {?|undefined} */
    Options.prototype.forceVisible;
}
